doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -4,6 +4,87 @@
  Proof handling
 -------------------
 
+.. todo PR: my example is pretty bad--can you suggest a better one?
+   Also need to position/integrate this better
+
+Proof State
+-----------
+
+Proof state consists of one or more subgoals that need to be proven.
+Each subgoal has a :gdef:`conclusion` (the statement that is to be proven)
+and may have zero or more :gdef:`hypotheses <hypothesis>`,
+which are specific facts that can be used to proving the conclusion (in
+addition to definitions, proven theorems, etc.).
+
+Hypotheses appear above the line and the conclusion appears below the line.
+Both of them are types.  Hypotheses have names, which appear before the colon
+and may be referred to in the conclusion or other hypotheses.
+Some hypotheses are akin to specifying the type of a variable, for example `n : nat`.
+
+The term ":gdef:`goal`" may refer to an entire subgoal or to the conclusion
+of a subgoal, depending on the context.
+
+.. todo PR where does the term "hypothesis" come from?  It is not the
+   common definition used in science.  Does it come from elsewhere or
+   it Coq's invention to use it this way?  I could observe that it's
+   a misnomer
```

Comment:
It's just the "hypothesis" from math as opposed to the rest of science (see the disambiguation on top of the Wikipedia page: https://en.wikipedia.org/wiki/Hypothesis).